internal/cmd/modules/modules.go

```diff
@@ -89,3 +114,54 @@ func listAllModules() ([]string, clierror.Error) {
 	}
 	return out, nil
 }
+
+func listManagedModules(cfg *modulesConfig) ([]string, clierror.Error) {
+	GVRKyma := schema.GroupVersionResource{
+		Group:    "operator.kyma-project.io",
+		Version:  "v1beta2",
+		Resource: "kymas",
+	}
+
+	unstruct, err := cfg.KubeClient.Dynamic().Resource(GVRKyma).Namespace("kyma-system").Get(cfg.Ctx, "default", metav1.GetOptions{})
+	if err != nil {
+		return nil, clierror.Wrap(err, clierror.New("while getting Kyma CR"))
+	}
+
+	moduleNames, err := getModuleNames(unstruct)
+	if err != nil {
+		return nil, clierror.Wrap(err, clierror.New("while getting module names from CR"))
+	}
+
+	return moduleNames, nil
+}
+
+func getModuleNames(unstruct *unstructured.Unstructured) ([]string, error) {
+	var moduleNames []string
+	managedFields := unstruct.GetManagedFields()
+	for _, field := range managedFields {
+		var data map[string]interface{}
+		err := json.Unmarshal(field.FieldsV1.Raw, &data)
+		if err != nil {
+			return nil, err
+		}
+
+		spec, ok := data["f:spec"].(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		modules, ok := spec["f:modules"].(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		for key := range modules {
+			if strings.Contains(key, "name") {
+				name := strings.TrimPrefix(key, "k:{\"name\":\"")
```

Comment:
Is it safe to use string operation on such structures? What will be done for string `"k : { "name" : "abcd" }` (very similar but with spaces)? Maybe we are sure that this string always ha fixed structure and this solution is ok?

Reply:
There could never be a space in there so we think it is okay